### PR TITLE
fix: ContactForm no longer show old contact after submit

### PR DESCRIPTION
**Pull request by JF-Cozy**
At the time of the creation/editing of a contact, when a field was modified, during the validation of the form, the old data was re-displayed (because of re-rendering the component with old data)

### CODE REVIEW

**Comment by y-lohse** [comment]
Quick question:

> the old data was re-displayed (because of re-rendering the component with old data)

Why isn't it possible to re-render the component with data that's up to date ?

**Review by y-lohse — COMMENTED** [review]
The fix looks good to me but I find the code is becoming hard to follow between ContactForm.onSubmit, submitForm and setFormPending. I think maybe grouping things in a single `handleSubmit` function would make the flow clearer?

**Comment by JF-Cozy** [comment]
> Quick question:
> 
> > the old data was re-displayed (because of re-rendering the component with old data)
> 
> Why isn't it possible to re-render the component with data that's up to date ?

Because the data comes from the properties, and if the rendering is done before the form is submitted (which was the case before) the data coming from the properties is not yet updated.

**Comment by JF-Cozy** [comment]
We can't have only one `handleSubmit` but after discussion, here's the new clearer code :)

**Review by y-lohse — APPROVED** [review]
Thanks, much clearer 👍
